--- runtime-decompiler/src/main/java/org/jrd/backend/core/DecompilerRequestReceiver.java
+++ runtime-decompiler/src/main/java/org/jrd/backend/core/DecompilerRequestReceiver.java.nw
@@ -214,9 +214,9 @@
         list.removeAll(Arrays.asList("", null));
         List<String> list1 = new ArrayList<>();
         for (String s : list) {
-            if (!(s.contains("Lambda") | s.startsWith("["))){
+//            if (!(s.contains("Lambda") | s.startsWith("["))){
                 list1.add(s);
-            }
+//            }
         }
         java.util.Collections.sort(list1);
         return list1.toArray(new String[]{});

